### ISSUE TITLE
I need to pass sequence without `vit_tokens`, even though `visual_und = True`

### ISSUE THREAD

**Issue by gracious-patience** [ByteDance-Seed/Bagel#137]
Hi, guys! I rewritten dataloaders for my infrastructure, so now i can have purely `visual_gen` samples in the sequence, even though i have inited the whole model with SigCLIP. In this case bagel.py returns error that `vit_token_seqlens` are None. However, if I skip this
```
cu_seqlens = torch.nn.functional.pad(torch.cumsum(vit_token_seqlens, dim=0), (1, 0))
```
 part in forward , the model freezes. How can i update the code, so i can handle this case? In my situation I can't guarantee  I'll have certain tasks in my sequence, so I need to handle it like this somehow :)

**Comment by Lane-Chang** [comment]
I have met the same bug. Have you solved it?

**Comment by gracious-patience** [comment]
Hi! The reason of freezing while having the multitask training is FSDP. When some workers receive both understanding and generation tasks and other, e g only generation (due to cfg drops and small weights for understanding, for instance), then different workers might use not all network submodules, and FSDP stucks in a deadlock while synchronising among all workers.

Personally I ended up with 2 possible solutions:

1. Either somehow ensuring all tasks are presented in a batch. You can do this using larger max_seq_len and lower probabilities of conditions drops
2. Or stop using different tasks in the same batch and rather change tasks in different iterations. I tried myself this way while finetuning and got the same results compared to multitasking in the batch.

**Comment by CostaliyA** [comment]
the same
